challenge_3/cpp/blakebarnes00/src/main.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+void print_array(std::vector<int> array)
+{
+  for(int i = 0; i < array.size(); i++)
```

Comment:
Can you explain how that loop works? I have seen it more recently and I can only presume that it removes the middle part of the for loop?

If that is what it does, how would you do it if you are doing i-- instead?